File: aspire-starter/aspire-starter.ApiService/Models/Patient.cs
using System.ComponentModel.DataAnnotations;

namespace aspire_starter.ApiService.Models;

public class Patient
{
    public int Id { get; set; }

    [Required, MaxLength(50)]
    public string FirstName { get; set; } = string.Empty;

    [Required, MaxLength(50)]
    public string LastName { get; set; } = string.Empty;

    public DateTime? DateOfBirth { get; set; }

    [MaxLength(10)]
    public string? Gender { get; set; }

    [Required, MaxLength(20)]
    public string Phone { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Email { get; set; }

    [MaxLength(500)]
    public string? Address { get; set; }

    [MaxLength(5)]
    public string? BloodGroup { get; set; }

    [MaxLength(100)]
    public string? EmergencyContactName { get; set; }

    [MaxLength(20)]
    public string? EmergencyContactPhone { get; set; }

    public DateTime RegistrationDate { get; set; } = DateTime.UtcNow;

    public bool IsActive { get; set; } = true;

    public ICollection<Appointment> Appointments { get; set; } = [];
    public ICollection<MedicalRecord> MedicalRecords { get; set; } = [];
    public ICollection<Billing> Bills { get; set; } = [];
}

File: aspire-starter/flutter_app/lib/models/patient.dart
class Patient {
  final int id;
  final String firstName;
  final String lastName;
  final DateTime? dateOfBirth;
  final String? gender;
  final String phone;
  final String? email;
  final String? address;
  final String? bloodGroup;
  final String? emergencyContactName;
  final String? emergencyContactPhone;
  final DateTime registrationDate;
  final bool isActive;

  Patient({
    this.id = 0,
    required this.firstName,
    required this.lastName,
    this.dateOfBirth,
    this.gender,
    required this.phone,
    this.email,
    this.address,
    this.bloodGroup,
    this.emergencyContactName,
    this.emergencyContactPhone,
    DateTime? registrationDate,
    this.isActive = true,
  }) : registrationDate = registrationDate ?? DateTime.now();

  String get fullName => '$firstName $lastName';

  factory Patient.fromJson(Map<String, dynamic> json) => Patient(
    id: json['id'] as int? ?? 0,
    firstName: json['firstName'] as String? ?? '',
    lastName: json['lastName'] as String? ?? '',
    dateOfBirth: json['dateOfBirth'] != null ? DateTime.parse(json['dateOfBirth'] as String) : null,
    gender: json['gender'] as String?,
    phone: json['phone'] as String? ?? '',
    email: json['email'] as String?,
    address: json['address'] as String?,
    bloodGroup: json['bloodGroup'] as String?,
    emergencyContactName: json['emergencyContactName'] as String?,
    emergencyContactPhone: json['emergencyContactPhone'] as String?,
    registrationDate: json['registrationDate'] != null ? DateTime.parse(json['registrationDate'] as String) : DateTime.now(),
    isActive: json['isActive'] as bool? ?? true,
  );

  Map<String, dynamic> toJson() => {
    if (id > 0) 'id': id,
    'firstName': firstName,
    'lastName': lastName,
    'dateOfBirth': dateOfBirth?.toIso8601String(),
    'gender': gender,
    'phone': phone,
    'email': email,
    'address': address,
    'bloodGroup': bloodGroup,
    'emergencyContactName': emergencyContactName,
    'emergencyContactPhone': emergencyContactPhone,
    'isActive': isActive,
  };
}
